blocks/header/header.js

```diff
@@ -85,6 +89,107 @@ function toggleMenu(nav, navSections, forceExpanded = null) {
   }
 }
 
+async function fetchSearchData() {
+  if (!window.searchData) {
+    const resp = await fetch(`/${window.location.pathname.split('/')[1]}/query-index.json`);
+    const json = await resp.json();
+    window.searchData = json;
+  }
+  return window.searchData;
+}
+
+async function search(value) {
+  const searchData = await fetchSearchData();
+  const hits = [];
+  for (let i = 0; i < searchData.data.length; i += 1) {
+    const e = searchData.data[i];
+    const text = [e.title, e.description].join(' ').toLowerCase();
+    if (text.includes(value.toLowerCase())) {
+      hits.push(e);
+    }
+  }
+  return hits;
+}
+
+function decorateSearch(block) {
+  const searchSection = block.querySelector('a > .icon-search');
+  if (searchSection) {
+    const searchElement = document.createElement('div');
+    searchElement.classList.add('nav-search');
+    searchSection.parentElement.parentElement
+      .replaceChild(searchElement, searchSection.parentElement);
+    searchElement.appendChild(searchSection);
+
+    const aside = document.createElement('aside');
+    aside.classList.add('nav-search-aside');
+    aside.classList.add('nav-search-aside-empty');
+
+    const input = document.createElement('input');
+    input.classList.add('nav-search-input');
+    input.type = 'search';
+    input.value = new URL(window.location).searchParams.get('ee_search_query');
+
+    input.addEventListener('input', async (e) => {
+      const url = new URL(window.location);
+      if (e.target.value.length >= 3) {
+        url.searchParams.set('ee_search_query', e.target.value);
+        const hits = await search(e.target.value);
+        if (hits.length > 0) {
+          aside.classList.remove('nav-search-aside-empty');
+          aside.innerHTML = `<h1 class="nav-search-result-title">Search Results for: ${input.value}</h1>
```

Comment:
be careful with allowing user controlled input in conjunction with `innerHTML`, especially one that can be shared by URL

Reply:
Good point, I use a textContent now which should be better.